src/openrct2/peep/Guest.cpp

```diff
@@ -1585,30 +1587,34 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
         if (itemValue < price)
         {
             itemValue -= price;
-            if (shopItem == SHOP_ITEM_UMBRELLA)
+            if (shopItem == SHOP_ITEM_UMBRELLA && climate_is_raining())
             {
-                if (climate_is_raining())
-                    goto loc_69B221;
+                //goto loc_69B221; now using below variable to skip all the code
+                //until the place where loc_69B221 was;
+                isRainingAndUmbrella = true;
```

Comment:
Not sure if this is is the best way, sounds like a break (or an even better refactor) would make the code clearer

Reply:
I will work on a refactor; it seems like it could be rewritten to be clearer.